lib/matplotlib/tests/test_axes.py

```diff
@@ -1754,6 +1754,27 @@ def test_hist_datetime_datasets():
     ax.hist(data, stacked=False)
 
 
+@pytest.mark.parametrize("bins_preprocess",
+                         [lambda bins: None,
```

Comment:
I think that if you leave out this test case, you can actually assert that the bins are what you expect them to be, since all the other ones return valid representations of the underlying data and should thus set the histogram binning to the same values. Maybe move the `None` test to a separate test?

This test does ensure that the current failure mode (raising an error) doesn't occur, but it doesn't ensure that these different types actually work as expected, which would be nice.

Reply:
On closer inspection, the previous test implicitly covers the `None` case, so I think just removing the `None` case and then asserting that the bins are equal to `mpl.dates.date2num(date_edges)` is whats required, so I did that.